File: client/common/protocol/err_message.go
package protocol

import "errors"

type Err struct{}

func (err *Err) Serialize() []byte {
	return buildHeader([]byte{ERR_OP}, 0)
}

func (err *Err) Deserialize(stream []byte) error {
	if !compareStreams(err.Serialize(), stream) {
		return errors.New("invalid Err message")
	}
	return nil
}

func (err *Err) ShouldAck() bool {
	return false
}

File: server/common/protocol/err_message.py
from common.protocol.message import Message


class ErrMessage(Message):
    ERR_OP = bytes([0x03])
    

    def serialize(self) -> bytes:
        return self._build_header(self.ERR_OP, 0)
    
    def should_ack(self) -> bool:
        return False